fix: replace custom loading spinners with skeleton components

harsh04044 opened the pull request:
## Proposed Changes

Fixes #14749 

I found two places where custom loading spinners were being used instead of skeleton loaders. This was inconsistent with the rest of the codebase, which uses skeleton components for better UX. I've replaced both spinners with appropriate skeleton components:

- **FacilityDetailsPage**: Replaced the centered spinner with skeleton loaders that show the facility card structure and user cards grid layout
- **UserDepartmentsTab**: Replaced the centered spinner with `CardGridSkeleton` to match the department cards grid layout

The skeleton loaders provide better UX by showing users the content structure while data loads, which is consistent with how other pages in the app handle loading states.

**Screenshots:**
<img width="1451" height="859" alt="image" src="https://github.com/user-attachments/assets/9666f3b9-fbda-4dcf-b553-25196da2216e" />
<img width="1417" height="852" alt="image" src="https://github.com/user-attachments/assets/2da24d79-19df-4f76-a8b5-33b3f0da4c82" />


Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [x] Update [product documentation](https://docs.ohc.network). (N/A - internal code improvement)
- [x] Ensure that UI text is placed in I18n files. (N/A - no new UI text added)
- [x] Prepare a screenshot or demo video for the changelog entry and attach it to the issue. (Screenshots provided in issue)
- [ ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.
- [ ] Add or update Playwright tests for related changes

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Replaced spinners with skeleton headers and responsive skeleton card grids for User Departments and Facility Details to improve perceived loading.
  * Added a public endpoint to fetch schedulable facility users for listing available users at a facility.

* **Refactor**
  * Updated loading-state structure to better mirror final content layout while preserving existing error handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

harsh04044 committed:
fix: replace custom loading spinners with skeleton components

- Replace spinner in FacilityDetailsPage with skeleton loaders
- Replace spinner in UserDepartmentsTab with CardGridSkeleton
- Improves UX by showing content structure during loading
- Consistent with rest of codebase

abhimanyurajeesh reviewed (changes requested):
I had requested changes in the closed PR, but I don’t see them applied here. Please make the necessary changes.

harsh04044 committed:
fix: address maintainer reviews for skeleton loading states

- Remove LinkUserToDepartmentSheet from loading state in UserDepartmentsTab
- Remove back button and LoginHeader from loading state in FacilityDetailsPage
- Replace manual skeleton divs (bg-gray-200 animate-pulse) with Skeleton component
- Fix deprecated routes usage by replacing with facilityApi.getUsers
- Loading states now show only skeleton loaders without interactive elements

Addresses maintainer feedback:
- 'is this needed?' - removed unnecessary interactive elements from loading state
- 'here also' - removed back button and LoginHeader from loading state
- 'SkeletonLoading is for removing these right?' - replaced manual divs with Skeleton component

Fixes #14751

Jacobjeevan commented on src/pages/Facility/FacilityDetailsPage.tsx:
```diff
@@ -41,9 +43,9 @@ export function FacilityDetailsPage({ id }: Props) {
-    queryKey: [routes.getScheduleAbleFacilityUsers, id],
-    queryFn: query(routes.getScheduleAbleFacilityUsers, {
-      pathParams: { facility_id: id },
+    queryKey: ["facilityUsers", id],
```
wrong api, be sure to test when changing APIs.

Add getPublicScheduleableFacilityUsers under getPublicScheduleableFacilityUser in scheduleApi.ts and use that. 

cc: @nihal467

harsh04044 replied:
@Jacobjeevan  @nihal467  I've addressed this in my latest commit. I'll make sure to test API changes myself in the future. Thanks for the feedback

Jacobjeevan replied:
use diff query key. Something like "facilityUsersPublic" should work.

@harsh04044

harsh04044 replied:
@Jacobjeevan  i have chaged querykey in my latest commmit

yash-learner replied:
https://github.com/ohcnetwork/care_fe/pull/14880#discussion_r2745816049

harsh04044 committed:
fix: use correct API for public schedulable facility users

- Replace facilityApi.getUsers with scheduleApis.appointments.getPublicScheduleableFacilityUsers
- Add getPublicScheduleableFacilityUsers API definition to scheduleApi.ts
- Remove unused facilityApi import from FacilityDetailsPage.tsx

Addresses maintainer feedback on PR #14880

harsh04044 committed:
fix: use correct API for public schedulable facility users

- Replace facilityApi.getUsers with scheduleApis.appointments.getPublicScheduleableFacilityUsers
- Add getPublicScheduleableFacilityUsers API definition to scheduleApi.ts
- Remove unused facilityApi import from FacilityDetailsPage.tsx

Addresses maintainer feedback on PR #14880

harsh04044 committed:
Merge upstream/develop: resolve conflict in FacilityDetailsPage.tsx

- Accept upstream version using publicFacilityApi.listSchedulableUsers
- Remove unused scheduleApis import
- Use camelCase facilityId parameter for consistency

harsh04044 committed:
fix: remove trailing newline in FacilityDetailsPage.tsx

- Remove extra blank line to satisfy prettier lint rules
- Auto-fixed by prettier during lint-fix

abhimanyurajeesh commented on src/types/scheduling/scheduleApi.ts:
```diff
@@ -186,6 +186,11 @@ export default {
+      path: "/api/v1/facility/{facility_id}/schedulable_users/",
+      method: HttpMethod.GET,
+      TRes: Type<PaginatedResponse<UserReadMinimal>>(),
+    },
```
where is this used?

harsh04044 replied:
@abhimanyurajeesh 
This endpoint is used in `src/pages/Facility/FacilityDetailsPage.tsx` (lines 44-48) to fetch schedulable users for a facility. 

It was added to address feedback from @Jacobjeevan to replace the incorrect `facilityApi.getUsers` usage.

yash-learner replied:
@harsh04044 revert the change to latest develop 

In this commit https://github.com/harsh04044/care_fe/commit/83159d5489dac99ed4da8cc87f1bb1a28a90f80c#diff-adf19fecbf15ca8f915dfcbf4d6f2fe4d46b7e7ad5716056ad3f1bcecbfe34a4 you had changed to 

<img width="2947" height="562" alt="image" src="https://github.com/user-attachments/assets/cf351542-3af7-4dff-aaca-410158ba08e9" />

That accidental change had lead to Jeevan's comment.


----

On develop

https://github.com/ohcnetwork/care_fe/blob/develop/src/pages/Facility/FacilityDetailsPage.tsx

<img width="1192" height="342" alt="image" src="https://github.com/user-attachments/assets/c774465e-deb4-4567-bbf8-e884575ac07f" />

yash-learner committed:
Merge branch 'develop' into issues/replace-spinners-with-skeletons

harsh04044 committed:
address requested change

harsh04044 committed:
revert unused code

yash-learner committed:
Merge branch 'develop' into issues/replace-spinners-with-skeletons

yash-learner committed:
Merge branch 'develop' into issues/replace-spinners-with-skeletons
